jaworski_cellseg/configuration_widget.py

```diff
@@ -0,0 +1,151 @@
+        auto_call=False,
+        configuration_names={
+            "label": "Config",
+            "widget_type": "ComboBox",
+            "choices": configurations,
+        },
+        save_confg={
+            "label": "Save",
+        },
+    )
+    def configuration_widget(
+        configuration_names: str = configurations[current_config],
+        save_confg: Optional[str] = "",
+    ):
+
+        if configuration_widget.save_confg.value:
+            key = configuration_widget.save_confg.value
+            with open(config_file_path, "w") as file:
+                
+                jaworski_obj.config["settings"][
+                    configuration_widget.save_confg.value
+                ] = generate_configuration_data(jaworski_obj)
+                yaml.dump(jaworski_obj.config, file)
+                # Update dropdown
+                new_choices = list(configuration_widget.configuration_names.choices)
+                if key not in new_choices:
+                    new_choices.append(key)
+                    configuration_widget.configuration_names.choices = new_choices
+                    configuration_widget.configuration_names.value = key
+
+    def update_layer_dropdown(event, jaworski_obj):
+        configuration = jaworski_obj.config["settings"][event]
+        pre_process_mappings = {
+            "gaussian_checkbox.value": "gausianFilter",
+            "gaussian_factor.value": "gaussian_sigma",
+            "contrast_adjustment.value": "contrast_adjustment",
+            "contrast_lower.value": "contrast_min",
+            "contrast_upper.value": "contrast_max",
+            "binary_map_threshold.value": "binary_map_threshold",
+        }
+
+        inferer_mappings = {
+            "use_window_choice.setChecked": "inference_use_window_choice",
+            "window_size_choice.setCurrentIndex": "inference_window_size",
+            "model_input_size.setValue": "inference_input",
+            "window_overlap_slider.setValue": "inference_overlap",
+            "thresholding_checkbox.setChecked": "inference_perform_threhold",
+            "thresholding_slider.setValue": "inference_perform_threhold_value",
+            "use_instance_choice.setChecked": "inference_instance_segmentation",
+        }
+
+        set_widget_values(
+            jaworski_obj.pre_process_data_widget, configuration, pre_process_mappings
+        )
+        set_widget_values(jaworski_obj.inferer_widget, configuration, inferer_mappings)
+        voronoi_widget = jaworski_obj.inferer_widget.instance_widgets.methods[
+            "Voronoi-Otsu"
+        ]
+        voronoi_widget.counters[0].setValue(
+            configuration["inference_instance_segmentation_spot_signma"]
+        )
+        voronoi_widget.counters[1].setValue(
+            configuration["inference_instance_segmentation_outline_signma"]
+        )
+        voronoi_widget.counters[2].setValue(
+            configuration["inference_instance_segmentation_small_object_removal"]
+        )
```

Comment:
You could use a for loop with `enumerate`:
```suggestion
        config_keys = [
            "inference_instance_segmentation_spot_signma",
            "inference_instance_segmentation_outline_signma",
            "inference_instance_segmentation_small_object_removal",
        ]

        for i, key in enumerate(config_keys):
            voronoi_widget.counters[i].setValue(configuration[key])
```

Reply:
Maybe move it to a function as it's needed elsewhere.

Reply:
Done